### PR TITLE
Update for new rclcpp_components

### PR DESCRIPTION
**Pull request by mjcarroll**
Modify the demos to work with the new composition mechanism from rclcpp_components.

Connects to ros2/rclcpp#665

### CODE REVIEW

**Comment by wjwwood** [comment]
Does this need to be updated at all? https://index.ros.org/doc/ros2/Tutorials/Composition/

**Comment by mjcarroll** [comment]
@wjwwood Yes docs need to be updated, I'm going to do it in a follow-up PR.

File: composition/include/composition/client_component.hpp [hunk 1]
```diff
@@ -26,7 +26,7 @@ class Client : public rclcpp::Node
 {
 public:
   COMPOSITION_PUBLIC
-  Client();
+  explicit Client(rclcpp::NodeOptions options);
```
Comment (dirk-thomas) [inline composition/include/composition/client_component.hpp]: Should this be a const reference as in the Node API?

Comment (dirk-thomas) [reply composition/include/composition/client_component.hpp]: See #323.

**Pull request merged**